Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewJSONDataProvider.swift

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct WhatsNewJSONDataProvider: WhatsNewDataProvider {
+    
+    let jsonData: Data?
+    private struct WhatsNewItemRoot: Decodable {
+        let items: [RemoteWhatsNewItem]
+    }
+
+    private struct RemoteWhatsNewItem: Decodable {
+        let imageURL: URL
+        let title: String
+        let subtitle: String
+    }
+
+    func getData() throws -> [WhatsNewItem] {
+        guard let jsonData else { return [] }
+                
+        do {
+            let decoder = JSONDecoder()
+            decoder.keyDecodingStrategy = .convertFromSnakeCase
+            let root = try decoder.decode(WhatsNewItemRoot.self, from: jsonData)
+            return root.items.map {
+                WhatsNewItem(imageURL: $0.imageURL,
+                                        title: $0.title,
+                                        subtitle: $0.subtitle)
+            }
+        } catch {
```

Comment:
I guess the error here should be just thrown since this method is already throwable, shouldn't it? If that is the case there would be no reason for `do catch`.

Reply:
Resolving as irrelevant for the current context considering that the JSON Provider will be reviewed anyway in the future. 🙏